### PR TITLE
Enable tagging of spot requests

### PR DESCRIPTION
**Pull request by ofosos**
This adds a new parameter to the EBS builders named `spot_tags`. This
parameter accepts a map of tags, much like `tags`. These tags will be
applied to a spot request that is created.

Improve visibility.

### CODE REVIEW

**Comment by ofosos** [comment]
Force push: rebase & move `spot_tags` into amazon/common/run_config.go

**Comment by ofosos** [comment]
Force-push: add a check around spot tag creation, this would error, when we had no spot tags defined, but were running a spot request.

**Comment by ofosos** [comment]
Did a check in prepare to validate with spot_tags set, spot_price is non-nil. Also rebase. Docs are next.

**Comment by SwampDragons** [comment]
hey @ofosos sorry for us taking so long to get back to this PR; if you want to rebase and fix conflicts I'll review and test it for merge.

**Comment by ofosos** [comment]
@SwampDragons I'll go ahead with this feature next week or over the weekend.

**Comment by ofosos** [comment]
@SwampDragons this does what I want, but I only tested the EBS builder.

**Comment by ofosos** [comment]
Hey @SwampDragons any comment?

File: builder/amazon/common/step_run_spot_instance.go [hunk 1]
```diff
@@ -227,6 +228,33 @@ func (s *StepRunSpotInstance) Run(ctx context.Context, state multistep.StateBag)
...
+	if err != nil {
+		err := fmt.Errorf("Error tagging spot request: %s", err)
+		state.Put("error", err)
+		ui.Error(err.Error())
+		return multistep.ActionHalt
+	}
+	spotTags.Report(ui)
+
+	if len(spotTags) > 0 && s.SpotTags.IsSet() {
+		// Retry creating tags for about 2.5 minutes
```
Comment (jen20) [inline builder/amazon/common/step_run_spot_instance.go]: How was the time window for this determined?

File: website/source/docs/builders/amazon-instance.html.md [hunk 2]
```diff
@@ -339,6 +339,10 @@ builder.
     best spot price. This must be one of: `Linux/UNIX`, `SUSE Linux`, `Windows`,
     `Linux/UNIX (Amazon VPC)`, `SUSE Linux (Amazon VPC)`, `Windows (Amazon VPC)`
 
+-   `spot_tags` (object of key/value strings) - Requires `spot_price` to
+    be set. This tells Packer to aplly tags to the spot request that is
```
Comment (jen20) [inline website/source/docs/builders/amazon-instance.html.md]: aplly => apply

File: website/source/docs/builders/amazon-ebsvolume.html.md [hunk 3]
```diff
@@ -264,6 +264,10 @@ builder.
     best spot price. This must be one of: `Linux/UNIX`, `SUSE Linux`, `Windows`,
     `Linux/UNIX (Amazon VPC)`, `SUSE Linux (Amazon VPC)` or `Windows (Amazon VPC)`
 
+-   `spot_tags` (object of key/value strings) - Requires `spot_price` to
+    be set. This tells Packer to aplly tags to the spot request that is
```
Comment (jen20) [inline website/source/docs/builders/amazon-ebsvolume.html.md]: aplly => apply

File: website/source/docs/builders/amazon-ebssurrogate.html.md [hunk 4]
```diff
@@ -336,6 +336,10 @@ builder.
     best spot price. This must be one of: `Linux/UNIX`, `SUSE Linux`, `Windows`,
     `Linux/UNIX (Amazon VPC)`, `SUSE Linux (Amazon VPC)`, `Windows (Amazon VPC)`
 
+-   `spot_tags` (object of key/value strings) - Requires `spot_price` to
+    be set. This tells Packer to aplly tags to the spot request that is
```
Comment (jen20) [inline website/source/docs/builders/amazon-ebssurrogate.html.md]: aplly => apply

File: website/source/docs/builders/amazon-ebs.html.md [hunk 5]
```diff
@@ -343,6 +343,10 @@ builder.
     best spot price. This must be one of: `Linux/UNIX`, `SUSE Linux`, `Windows`,
     `Linux/UNIX (Amazon VPC)`, `SUSE Linux (Amazon VPC)`, `Windows (Amazon VPC)`
 
+-   `spot_tags` (object of key/value strings) - Requires `spot_price` to
+    be set. This tells Packer to aplly tags to the spot request that is
```
Comment (jen20) [inline website/source/docs/builders/amazon-ebs.html.md]: aplly => apply

File: builder/amazon/common/step_run_spot_instance.go (hunk 1 above)
Comment (ofosos) [reply builder/amazon/common/step_run_spot_instance.go]: It's a common timeout throughout the codebase.

Comment (ofosos) [reply builder/amazon/common/step_run_spot_instance.go]: I'm fine to set it to any other value, but 2.5 minutes seems like a nice compromise that'll allow some time for the API call to go through - even in the worst case - while not waiting forever.

Comment (jen20) [reply builder/amazon/common/step_run_spot_instance.go]: That seems OK then.

**Comment by jen20** [comment]
Thanks @ofosos! This is now merged, I'll update the CHANGELOG.

**Pull request merged**